src/configparser.cpp

```diff
@@ -609,14 +609,14 @@ unsigned long ConfigTag::getDuration(const std::string& key, unsigned long def,
 	if (!readString(key, duration))
 		return def;
 
-	if (!InspIRCd::IsValidDuration(duration))
+	unsigned long ret = def;
```

Comment:
```suggestion
	unsigned long ret;
```

The value you assign to ret will never actually be used so you don't need a default.